diff --git a/node_modules/expo-updates/plugin/build/withUpdates.js b/node_modules/expo-updates/plugin/build/withUpdates.js
index 618c37f..4b548e8 100644
--- a/node_modules/expo-updates/plugin/build/withUpdates.js
+++ b/node_modules/expo-updates/plugin/build/withUpdates.js
@@ -8,6 +8,19 @@ const withUpdates = (config, props = {}) => {
     const expoUsername = (props || {}).expoUsername ?? (0, config_1.getAccountUsername)(config);
     config = config_plugins_1.AndroidConfig.Updates.withUpdates(config, { expoUsername });
     config = config_plugins_1.IOSConfig.Updates.withUpdates(config, { expoUsername });
+    let runtimeVersion;
+    if (typeof config.runtimeVersion === 'string') {
+        runtimeVersion = `string:${runtimeVersion}`;
+    }
+    else {
+        if (config.runtimeVersion?.policy === 'appVersion') {
+            runtimeVersion = `string:${config.version}`;
+        }
+        else {
+            runtimeVersion = config.runtimeVersion;
+        }
+    }
+    config.runtimeVersion = runtimeVersion;
     return config;
 };
 exports.default = (0, config_plugins_1.createRunOncePlugin)(withUpdates, pkg.name, pkg.version);
diff --git a/node_modules/expo-updates/plugin/src/withUpdates.ts b/node_modules/expo-updates/plugin/src/withUpdates.ts
index e094cab..97e2021 100644
--- a/node_modules/expo-updates/plugin/src/withUpdates.ts
+++ b/node_modules/expo-updates/plugin/src/withUpdates.ts
@@ -9,6 +9,18 @@ const withUpdates: ConfigPlugin<{ expoUsername?: string } | void> = (config, pro
 
   config = AndroidConfig.Updates.withUpdates(config, { expoUsername });
   config = IOSConfig.Updates.withUpdates(config, { expoUsername });
+  let runtimeVersion;
+  if (typeof config.runtimeVersion === 'string') {
+    runtimeVersion = `string:${runtimeVersion}`;
+  } else {
+    if (config.runtimeVersion?.policy === 'appVersion') {
+      runtimeVersion = `string:${config.version}`;
+    } else {
+      runtimeVersion = config.runtimeVersion;
+    }
+  }
+  config.runtimeVersion = runtimeVersion;
+
   return config;
 };
 
